spec/language/ruby_pattern_matching_spec.rb

```diff
@@ -1281,9 +1283,9 @@ def test_deconstruct_keys
     end
 
     # https://github.com/jruby/jruby/issues/7854
-    unless defined?(JRUBY_VERSION)
+    unless defined?(JRUBY_VERSION) || RUBY_ENGINE == "truffleruby"
     assert_block do
-      case {}
+      case C.new({})
       in {}
         C.keys == nil
       end
```

Comment:
I suppose it's a bug in https://github.com/ruby/ruby/blob/master/test/ruby/test_pattern_matching.rb that was hidden - `C.keys` returned keys captured in some previous test case, that by chance matches `C.new({})` and captures the `nil` value.

Reply:
Good catch! Yeah, having `case {} in ` and checking for `C.keys` doesn't make any sense here

Reply:
The fix in the Ruby repo - https://github.com/ruby/ruby/pull/10757